src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -542,33 +542,17 @@ void ImageReplayer<I>::on_start_fail_finish(int r)
   Context *on_stop_finish(nullptr);
   {
     Mutex::Locker locker(m_lock);
-    if (m_stop_requested) {
-      assert(r == -EINTR);
-      dout(20) << "start interrupted" << dendl;
-      m_state = STATE_STOPPED;
-      m_stop_requested = false;
+    if (r < 0) {
+      derr << "start failed: " << cpp_strerror(r) << dendl;
     } else {
-      assert(m_state == STATE_STARTING);
-      dout(20) << "start failed" << dendl;
-      m_state = (r < 0) ? STATE_UNINITIALIZED : STATE_STOPPED;
+      dout(20) << "start interrupted" << dendl;
```

Comment:
Minor: Previous this message was for the case, when stop is called before start is complete (on_start_fail_start(-EINTR)). Now it is for the case when bootstrap returns "remote image is non-primary or local image is primary". It would be nice still to be able to distinguish interrupted start from other errors.